### ISSUE TITLE
Provide an easier way to change the executable icon for projects exported to Windows

### ISSUE THREAD

**Issue by valfus** [godotengine/godot-proposals#1097]
**Describe the project you are working on:**
Currently nothing, just tests.

**Describe the problem or limitation you are having in your project:**
It is not possible to change the properties of the executable (for example the description of the file, the icon, etc.) at the time of export

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
By including rcedit in godot it will be possible to export the games without having to download it separately

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
I don't know

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
It is not possible

**Is there a reason why this should be core and not an add-on in the asset library?:**
because this way you can change the properties of the executable without having to download it separately to get everything ready. Then I believe that if the rcedit code were included in godot there would no longer be any need for wine on macOS / linux.

**Comment by Calinou** [comment]
rcedit can't be built as a cross-platform library, since it uses Windows-specific APIs. Its executable also requires WINE to be run on macOS/Linux. It's also not really maintained anymore by the looks of it: https://github.com/electron/rcedit/issues/91

**Comment by valfus** [comment]
Okay, couldn't it be included only for Windows until a cross-platform solution is found?
I would like to try to help but I don't know C / C ++. The only programming language that I know a little is Python, but I don't know if it could be done with Python?

**Comment by Calinou** [comment]
@valfus No, as we intend to keep Godot as cross-platform as possible. Moreover, bundling a third-party executable might look sketchy to people who are downloading and trying out Godot for the first time.

**Comment by pajcek11** [comment]
rcedit aside, as you have given me a good explanation on why it is needed, 
could someone please address my idea of just replacing the Godot icon?

1. there is a "fallback" Godot icon present in every project you export for windows. 
2. when you run an .exe where rcedit was not applied or it failed, the icon shown is a Godot icon
3. when I tried to hack the binaries to replace the icon (which I managed, but then the .exe wouldn't work because the embedded pkc got messed up), I saw that there is a proper .ico in there with 6 or 7 different resolutions of a Godot icon
4. therefore, why not just enable users to replace this "fallback"/default icon, which appears to work properly all the time (otherwise windows would give it some default executable icon)

5. And another question comes to mind - why can this default icon work properly, but everyone has to jump through hoops to import their custom one?

Thank you!

**Comment by Calinou** [comment]
> And another question comes to mind - why can this default icon work properly, but everyone has to jump through hoops to import their custom one?

The default icon is compiled into the executable at build-time. Doing so is easier and more reliable than changing the icon *after* it has been compiled.

That said, I think we can reasonably address this issue for now by making export templates not use any icon. The end result will be less problematic compared to every export template using the Godot icon by default.

**Comment by pajcek11** [comment]
>The default icon is compiled into the executable at build-time.
So, does this default icon exist somewhere physically in some directory (and if yes in which), or is it stored as bits somewhere in some file, being relatively inaccessible?
Just asking for a friend. Wouldn't cross my mind to just physically alter the default icon file and go on my merry way ;P ;P

**Comment by Calinou** [comment]
> So, does this default icon exist somewhere physically in some directory (and if yes in which), or is it stored as bits somewhere in some file, being relatively inaccessible?

Replace [`platform/windows/godot.ico`](https://github.com/godotengine/godot/blob/master/platform/windows/godot.ico) before building export templates.
